modules/eks/karpenter/main.tf

```diff
@@ -116,7 +178,8 @@ module "karpenter" {
           "sqs:ReceiveMessage",
         ]
         resources = [
-          aws_sqs_queue.interruption_handler[0].arn
+          #arn:aws:sqs:region:account_id:queue_name
+          format("arn:%s:sqs:%s:%s:%s", local.partition, var.region, local.account_id, local.interruption_handler_queue_name)
```

Comment:
This is a nasty hack. Use `one(          aws_sqs_queue.interruption_handler[*].arn)` and use [`iam_policy`](https://github.com/cloudposse/terraform-aws-iam-policy/blob/ee1a7314b45241e96f0d1e0c9e6745da6d088430/variables.tf#L1) instead of `iam_policy_statements`.

Reply:
tested and this worked too. Updated to use `iam_policy`